pkg/infra/iac2/templates_compiler.go

```diff
@@ -75,7 +75,7 @@ func (s stringTemplateValue) Raw() interface{} {
 
 func (v nestedTemplateValue) Parse() (string, error) {
 	childVal := v.childValue
-	return v.tc.resolveStructInput(&v.parentValue, childVal, v.useDoubleQuotedStrings, v.iacTag)
```

Comment:
I don't really understand what's going on here. You pass in the `*ResourceCreationTemplate`, and then it looks like you modify it but don't ever use it. But you never do anything with this original reference, so where does that information all go? What uses the template you build up?

(This is different from the unit test, where you first create an RCT, then pass it into this method, and then check that it got built up correctly.)

Reply:
ill add some docs as well, but the template has an appliedOutputs list now and we can add to that list if we recognize fields within the returned resource need to be applied in order for it to work properly.